src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_pix2pix_zero.py

```diff
@@ -0,0 +1,843 @@
+                [original paper](https://arxiv.org/abs/2302.03027). Used in discovering the edit direction.
+            target_embedding_path (`str`, defaults to None):
+                Local filepath to the embeddings of the target concept. Generation of the embeddings as per the
+                [original paper](https://arxiv.org/abs/2302.03027). Used in discovering the edit direction.
+            output_type (`str`, *optional*, defaults to `"pil"`):
+                The output format of the generate image. Choose between
+                [PIL](https://pillow.readthedocs.io/en/stable/): `PIL.Image.Image` or `np.array`.
+            return_dict (`bool`, *optional*, defaults to `True`):
+                Whether or not to return a [`~pipelines.stable_diffusion.StableDiffusionPipelineOutput`] instead of a
+                plain tuple.
+            callback (`Callable`, *optional*):
+                A function that will be called every `callback_steps` steps during inference. The function will be
+                called with the following arguments: `callback(step: int, timestep: int, latents: torch.FloatTensor)`.
+            callback_steps (`int`, *optional*, defaults to 1):
+                The frequency at which the `callback` function will be called. If not specified, the callback will be
+                called at every step.
+
+        Examples:
+
+        Returns:
+            [`~pipelines.stable_diffusion.StableDiffusionPipelineOutput`] or `tuple`:
+            [`~pipelines.stable_diffusion.StableDiffusionPipelineOutput`] if `return_dict` is True, otherwise a `tuple.
+            When returning a tuple, the first element is a list with the generated images, and the second element is a
+            list of `bool`s denoting whether the corresponding generated image likely represents "not-safe-for-work"
+            (nsfw) content, according to the `safety_checker`.
+        """
+        # 0. Define the spatial resolutions.
+        height = height or self.unet.config.sample_size * self.vae_scale_factor
+        width = width or self.unet.config.sample_size * self.vae_scale_factor
+
+        # 1. Check inputs. Raise error if not correct
+        self.check_inputs(
+            prompt,
+            self.conditions_input_image,
+            image,
+            source_embedding_path,
+            target_embedding_path,
+            callback_steps,
+            prompt_embeds,
+        )
+        if self.conditions_input_image and prompt_embeds:
+            logger.warning(
+                f"You have set `conditions_input_image` to {self.conditions_input_image} and"
+                " passed `prompt_embeds`. `prompt_embeds` will be ignored. "
+            )
+
+        # 2. Generate a caption for the input image if we are conditioning the
+        # pipeline based on some input image.
+        if self.conditions_input_image:
+            caption, preprocessed_image = generate_caption(image, self._captioner, self._captioner_processor)
```

Comment:
instead passing the model and processor here, can it become it become the class method ?